source/Octopus.Client/Repositories/Async/MixedScopeBaseRepository.cs

```diff
@@ -41,16 +42,18 @@ void ValidateThatICanUseACustomSpaceContext()
             }
         }
 
-        protected override void CheckSpaceResource(IHaveSpaceResource spaceResource)
+        protected override async Task CheckSpaceResource(IHaveSpaceResource spaceResource)
         {
-            Repository.Scope.Apply(
+            await Repository.Scope.Apply(
```

Comment:
[nit]

Same as above - could just `return` the task directly instead of awaiting.